### PR TITLE
create start contributing guide

### LINKED ISSUE #6845: [CCI] Update Contributing guide - help newers to make first PR
**Note**: issue to speed up the entry threshold of the contribution process

### Description:

This markdown file should help all new contributors. And should describe the whole process:
from introduction to the first pulll-request.

And describes issues such as:

- How to work with Git: 

  - How to create commits
  - How to create PRs

### PR DESCRIPTION
**Pull request by Nicksqain**
### Description
This is an updated guide that is ideal for all beginners.
It contains all the information beginners need.

I'm waiting for feedback :)

### Issues Resolved
#6845 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Nicksqain** [comment]
I would like to add this guide to all repositories changing the context
What do you think about it?

**Comment by reta** [comment]
@Nicksqain we already have a contributing guide (CONTRIBUTING.md, along with DEVELOPER_GUIDE.md), I don't think we need yet another one, may be filling the gaps in the existing ones would be helpful.

**Comment by Nicksqain** [comment]
@reta I myself have recently started contributing to opensearch project repositories. And when I started it was difficult for me to search for information. As a member of the CCI programme I would like to see the guide do just that. From less to more :)

**Comment by reta** [comment]
> And when I started it was difficult for me to search for information.

That would not make it easier - we would end up with 3 places having more or less the same information. We better should provide clear pointers where the required information lives. With that being said, OpenSearch landing page has the link to contribution guide right at the beginning: 
 - have you found it difficult to find the contribution guidelines?
 - have you found it difficult to follow the contribution guidelines?
 - have you found it difficult to configure your development environment? 
 - have you found missing links which prevented you to prepare everything  you needed to start contributing?

**Comment by Nicksqain** [comment]
@reta 
The entire guide was compiled from the responses of programme participants who were never able to make their first pull request. The problem was that they found it difficult to understand the guide itself. There is a lot of information and it all needs to be processed

**Comment by andrross** [comment]
@Nicksqain Thanks for getting involved in the project! I do agree with @reta here that the existing guides seem like the right place for this content.

We also have a readme at the [main landing page](https://github.com/opensearch-project) that links to an [ONBOARDING guide](https://github.com/opensearch-project/.github/blob/main/ONBOARDING.md). I would suggest going through those as well and look at adding/improving the content there.

**Comment by rednaksi91** [comment]
@owaiskazi19 @kotwanikunal @Poojita-Raj @Rishikesh1159 one of the CCI contributors went an extra mile to collect and compile all of the obstacles raised by the new contributors. He then went ahead and created comprehensive material that would make it easier for new contributors to get started. Let's take a closer look at what @Nicksqain has put together and incorporate it in either the Contributing or Developer guides.
What is the specific feedback for @Nicksqain ?

**Review by Poojita-Raj — COMMENTED** [review]
I agree with @reta that creating another guide called STARTING would not be the ideal solution in this case since we do clearly provide a contributing and developer guide. 

That being said, since you have shared that it is the experience of multiple contributors that they're hard to understand, it would be a good idea to go through the content of this starting guide and incorporate it into the existing guides to make them more user-friendly to beginners.

**Comment by dbwiddis** [comment]
> I agree with @reta that creating another guide called STARTING would not be the ideal solution in this case since we do clearly provide a contributing and developer guide.
> 
> That being said, since you have shared that it is the experience of multiple contributors that they're hard to understand, it would be a good idea to go through the content of this starting guide and incorporate it into the existing guides to make them more user-friendly to beginners.

I'll pile on with this recommendation.  I think documenting these experiences and obstacles is great, but I'm not sure another .md file is the best place.  This seems like something which could be written up as a blog post or forum post, however.

**Comment by owaiskazi19** [comment]
+1 to @reta and @Poojita-Raj. @Nicksqain thanks for bringing this up. Looking at the changes looks like the changes would fit in CONTRIBUTING.md. Let's add the changes there and we can review the new PR.

**Comment by Nicksqain** [comment]
@owaiskazi19
 Thank you all for the answer! I'll take all the comments into account. I think it's necessary to make a topic on the forum on this file. And for contributing.md I will extract all the most important and update it. Then I think it's worth changing the issue and continuing to work on changing the current guide.

 After I change contributing.md and will you accept the changes I will need to make pull requests to all repositories with this file?

File: STARTING.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+## Contributing
+
+### What actions contribute
+
+#### Issues
+
+**Issues** is a functionality on the GitHub platform that allows users to track and manage tasks, bugs, feature requests and <br> other project-related issues. It provides a collaborative environment where developers can discuss and solve problems related to a particular repository.
+
+_Note_: Since you are a beginner, we advise you to start with simple issues.
```
Comment (Poojita-Raj) [inline STARTING.md:63]: Good callout to bring attention to the good first issue label. This is definitely something we can incorporate into our current guides.

**Review by dblock — CHANGES_REQUESTED** [review]
We already have https://github.com/opensearch-project/.github/blob/main/ONBOARDING.md for org-wide onboarding, and it's linked from https://github.com/opensearch-project README. 

1. Make improvements to that with some content from here.
2. Link ONBOARDING in various CONTRIBUTING across the org if it was difficult to find?

**Pull request closed without merging**